src/js/graphql_endpoint/modules/root_query.ts

```diff
@@ -1,372 +1,341 @@
+import {printSchema} from "graphql/utilities";
 import {
-	GraphQLObjectType,
-	GraphQLInt,
-	GraphQLString,
-	GraphQLList,
-	GraphQLSchema,
-	GraphQLNonNull,
+    GraphQLObjectType,
+    GraphQLInt,
+    GraphQLString,
+    GraphQLList,
+    GraphQLSchema,
+    GraphQLNonNull,
 } from "graphql";
 
-import { LensNodeType, builtins } from "./schema";
+import { BaseNode } from "./schema";
 
 import {
-	getDgraphClient,
-	DgraphClient,
-	RawNode,
-	EnrichedNode,
+    getDgraphClient,
+    DgraphClient,
+    RawNode,
+    EnrichedNode,
 } from "./dgraph_client";
+import { Schema, SchemaClient } from "./schema_client";
+import { allSchemasToGraphql } from "./schema_to_graphql";
 
 type MysteryParentType = never;
 
 const getLenses = async (
-	dg_client: DgraphClient,
-	first: number,
-	offset: number
+    dg_client: DgraphClient,
+    first: number,
+    offset: number
 ) => {
-	console.debug("first, offset parameters in getLenses()", first, offset);
-
-	const query = `
-		query all($a: int, $b: int)
-		{
-			all(func: type(Lens), first: $a, offset: $b, orderdesc: score)
-			{
-				lens_name,
-				score,
-				node_key,
-				uid,
-				dgraph_type: dgraph.type,
-				lens_type,
-				scope {
-					uid,
-					node_key,
-					dgraph_type: dgraph.type,
-				}
-			}
-		}
-	`;
-
-	console.debug("Creating DGraph txn in getLenses");
-
-	const txn = dg_client.newTxn();
-
-	try {
-		console.debug("Querying DGraph for lenses in getLenses");
-		const res = await txn.queryWithVars(query, {
-			$a: first.toString(),
-			$b: offset.toString(),
-		});
-		console.debug("Lens response from DGraph", res);
-		return res.getJson()["all"];
-	} catch (e) {
-		console.error("Error in DGraph txn getLenses: ", e);
-	} finally {
-		console.debug("Closing Dgraph Txn in getLenses");
-		await txn.discard();
-	}
+    console.debug("first, offset parameters in getLenses()", first, offset);
+
```

Comment:
Unfortunately andrea and I got into a formatting war here. I'm gonna add a `make format` for TS asap.